### PR TITLE
Refactor quickstarts

### PR DESCRIPTION
**Pull request by adriendelsalle**
### What does this PR do?

This is a proposition of refactoring for the `quick-starts` part of the documentation, especially articles concerning **on-premise** clusters deployments currently merged in a single `deploy locally` page. 

I followed @l0rd [suggestion](https://github.com/eclipse/che/issues/16668#issuecomment-620551040) to sort methods as follow:
- local single-node cluster
- private **on-premise** production capable cluster
- cloud provided cluster

Even if I think it's more accurate to split this way:
- on-premise cluster
  - single-node only
  - prod capable
- cloud provided cluster

### What issues does this PR fix or reference?

After testing several methods to deploy Che (minikube, etc.), I found IMHO that the `deploy locally` page contains to much information. For a given tutorial, Minikube for example, a reader has  to find the next step for his deployment in a huge shortcuts/titles area on the right side accross non-relevant articles for his needs.

It is related to this [issue](https://github.com/eclipse/che/issues/16668) about creating a documentation for a bare metal Che deployment using Kubespray and its integration in the current doc. 

The `Che on bare metal using Kubespray` documentation was already submitted in this [PR](https://github.com/eclipse/che-docs/pull/1252). This new PR extends the previous one.

### Specify the version of the product this PR applies to. 

Eclipse Che 7.*

### CODE REVIEW

File: src/main/pages/che-7/overview/assembly_che-quick-starts.adoc [hunk 1]
```diff
@@ -20,22 +20,55 @@ summary:
...
 
-* link:{site-baseurl}che-7/accessing-che-from-openshift-developer-perspective/[Accessing {prod-short} from OpenShift Developer Perspective]
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-docker-desktop/[Docker Desktop]
+
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-minikube/[Minikube]
+
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-minishift/[Minishift]
+
+
+== On a private (__on-premise__) cluster
```
Comment (l0rd) [inline src/main/pages/che-7/overview/assembly_che-quick-starts.adoc]: I would say "On a private cloud" or "On an on-premise cluster". With a preference for the first one (c.f. hybrid clouds definitions from [microsoft](https://azure.microsoft.com/en-us/overview/what-is-hybrid-cloud-computing/) and [red hat](https://www.redhat.com/en/topics/cloud-computing/what-is-hybrid-cloud))

File: src/main/pages/che-7/overview/assembly_che-quick-starts.adoc [hunk 2]
```diff
@@ -20,22 +20,55 @@ summary:
...
+
+NOTE: `production-capable` means you can deploy clusters from __test-like__ xref:on-a-local-single-node-cluster[local single-node clusters] to production grade high-availability ones depending on your configuration (number of nodes, etc.)
+
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-bare-metal-using-kubespray/[Bare metal] __using Kubespray__
+
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-openshift-3-using-the-operator/[OpenShift 3] __using the Operator__
+
+* link:{site-baseurl}che-7/installing-{prod-id-short}-on-openshift-4-using-operatorhub/[OpenShift 4] __using OperatorHub__
+
+== On a cloud provided cluster
```
Comment (l0rd) [inline src/main/pages/che-7/overview/assembly_che-quick-starts.adoc]: For the same reason mentioned above I would change here to "On a public cloud provider"

**Comment by l0rd** [comment]
Nice work @adriendelsalle!

File: src/main/pages/che-7/overview/assembly_che-quick-starts.adoc (hunk 1 above)
Comment (adriendelsalle) [reply src/main/pages/che-7/overview/assembly_che-quick-starts.adoc]: Thanks for reviewing this PR so quickly! I have improved the wording as suggested.
I'm currently learning about all these things. 

The reason why I was talking about clusters locations more than cloud location is : can we consider a bare metal cluster (without virtualization) as a private cloud?
In the end, I agree the purpose is the same by providing *services*.

File: src/main/pages/che-7/overview/assembly_che-quick-starts.adoc (hunk 2 above)
Comment (adriendelsalle) [reply src/main/pages/che-7/overview/assembly_che-quick-starts.adoc]: fixed, thanks @l0rd

**Comment by adriendelsalle** [comment]
Hi @l0rd! I made modifications as you requested and #1263, first part of this PR concerning bare metal quickstart article using kubespray, was merged today (yipee! Thanks @boczkowska!).
Is it possible to review please?

**Comment by sunix** [comment]
test it:
https://che.openshift.io/factory?url=https://github.com/eclipse/che-docs/&override.projects.che-docs.source.location=https://github.com/adriendelsalle/che-docs&override.projects.che-docs.source.branch=refactor-quickstarts

**Comment by adriendelsalle** [comment]
Hi @sunix @rkratky 
Is it possible to review this PR please?
Thanks!

**Review by rkratky — CHANGES_REQUESTED** [review]
Review (PR) with suggestion provided at https://github.com/adriendelsalle/che-docs/pull/1

**Comment by adriendelsalle** [comment]
Closing this one in favor of #1330.
Minor adjustements could be done after the Antora migration.

Thanks!

**Pull request closed without merging**